### PR TITLE
Created fire_upon and fired_upon methods for cell class and test

### PR DESCRIPTION
**Pull request by amsmyth1**
Cell class and test code
Created methods: fire_upon, fired_upon

### CODE REVIEW

File: lib/cell.rb [hunk 1]
```diff
@@ -1,9 +1,10 @@
 class Cell
-  attr_reader :coordinate, :ship
+  attr_reader :coordinate, :ship, :fired_upon
```
Comment (jbranham1) [inline lib/cell.rb]: This is partly my bad but could we add each attr_reaader to its own line?

**Review by jbranham1 — CHANGES_REQUESTED** [review]
Looks good! I had some feedback on attr_readers that is partly my fault!

File: lib/cell.rb (hunk 1 above)
Comment (amsmyth1) [reply lib/cell.rb]: Good catch. Revised attr_reader to multiline syntax.

**Review by jbranham1 — APPROVED** [review]
Great job!

**Pull request merged**